### PR TITLE
Update for v2.2207

### PR DESCRIPTION
**Pull request by kburns**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by kburns** [comment]
@conda-forge-admin, please rerender

**Comment by kburns** [comment]
Hi @mxr-conda @ndmaxar @oblute @rluria14, I think this might be ready to go -- it's now building successfully with more stringent tests on linux and osx64 with openmpi and mpich.  I'm the lead developer of the underlying package and I'm eager to maintain and extend the support of the CF feedstock going forward.  Thanks!

**Comment by kburns** [comment]
@conda-forge/core could anyone take a quick look? I'm eager to start maintaining an extending the feedstock for this package.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
 {% set mpi_prefix = "mpi_" + mpi %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 7b6005c6c9b3b85c42da9a2ae85219b2722e2f61bda714d2518c4c058fd5d8df
+  url: https://files.pythonhosted.org/packages/6d/89/0f5b60cfd1d8dbe89f1f7293ee388ceef66f537ee8b5988cb6c7a8d35c6b/dedalus-2.2207.tar.gz
```
Comment (Tobias-Fischer) [inline recipe/meta.yaml]: Why was the url structure changed? With this change, the conda-forge bot won't detect version updates.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
+  sha256: 158adebf4b4939bb3d322dc81176965997f84a9ce54e020210db52eda4773dc0
 
 build:
-  number: 4
-  skip: true  # [win or py>37]
+  number: 0
+  skip: true  # [win or py<35]
```
Comment (Tobias-Fischer) [inline recipe/meta.yaml]:
```suggestion
  skip: true  # [win]
```

File: recipe/meta.yaml (hunk 1 above)
Comment (kburns) [reply recipe/meta.yaml]: It seems like it took a day for two for the short URL's to be updated by pypi, and this was failing when I first updated the recipe.  But it looks like that link is working now, so I changed it back in d598b7d.

File: recipe/meta.yaml (hunk 2 above)
Comment (kburns) [reply recipe/meta.yaml]: I think the underlying package has a hard 3.5+ requirement for Python, but maybe we don't need that here?  I can see CF is only building for 3.7+, but I didn't know if that's a guarantee / documented somewhere.

Comment (Tobias-Fischer) [reply recipe/meta.yaml]: Yes, correct, we don't build for these old versions of Python anymore, so this version constraint is implicit anyhow.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -22,39 +21,46 @@ requirements:
     - pip
     - python
     - setuptools
-    - cython
-    - mpi4py
-    - numpy
+    - cython >=0.22
+    - mpi4py >=2.0.0
+    - numpy >=1.20.0
```
Comment (Tobias-Fischer) [inline recipe/meta.yaml:26]: I'm not sure about the numpy pin. 

Typically we pin a certain version of numpy to each version of python (see https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy). 

As this package has a run_compatible, and now needs at least numpy 1.20, but most other packages are built against numpy 1.19, this leads to incompatibilities.

Just want to point that out, not sure whether anything needs changing.

Comment (kburns) [reply recipe/meta.yaml:26]: Yes, I wasn't too sure what to do here and tried to follow the directions as well as I could. But it looks like the pin-compatible description allows any numpy above 1.19 and below 2 for the run requirements, so other packages used that plain specification should be compatible with the 1.20+ that this package requires at runtime, right? Or could the issue be incompatibilities here during the build? The builds all seem to be passing now.

Comment (kburns) [reply recipe/meta.yaml:26]: @Tobias-Fischer think this is ok for now?

Comment (Tobias-Fischer) [reply recipe/meta.yaml:26]: Looks good to me, but I don't have merge rights. Sorry to ping @conda-forge/core

**Comment by pkgw** [comment]
Nice to see you here, Keaton! Thanks for putting in the work on this.

**Pull request merged**